Assets/MixedRealityToolkit.Tests/PlayModeTests/InputEventSystemTests.cs

```diff
@@ -0,0 +1,300 @@
+            inputSystem.RaisePointerClicked(inputSystem.GazeProvider.GazePointer, MixedRealityInputAction.None, 1);
+
+            Assert.AreEqual(objectBasedListener.pointerClickedCount, 1, "Pointer clicked event is not received by old API handler.");
+            Assert.Zero(objectBasedListener.pointerDownCount,           "Pointer down event is received by old API handler.");
+            Assert.Zero(objectBasedListener.pointerUpCount,             "Pointer up event is received by old API handler.");
+            Assert.Zero(objectBasedListener.pointerDraggedCount,        "Pointer dragged event is received by old API handler.");
+            Assert.Zero(objectBasedListener.speechCount,                "Speech event is received by old API handler.");
+            
+            // Wrong behavior, preserved for backward compatibility
+            Assert.AreEqual(handlerBasedListener.pointerClickedCount, 1, "Pointer clicked event is not received by new API handler.");
+            Assert.Zero(handlerBasedListener.pointerDownCount,           "Pointer down event is received by new  API handler.");
+            Assert.Zero(handlerBasedListener.pointerUpCount,             "Pointer up event is received by new API handler.");
+            Assert.Zero(handlerBasedListener.pointerDraggedCount,        "Pointer dragged event is received by new API handler.");
+            Assert.Zero(handlerBasedListener.speechCount,                "Speech event is received by new API handler.");
+            
+            Assert.AreEqual(handlerBasedListener1.pointerClickedCount, 1, "Pointer clicked event is not received by all-handlers component.");
+            Assert.Zero(handlerBasedListener1.pointerDownCount,           "Pointer down event is received by all-handlers component.");
+            Assert.Zero(handlerBasedListener1.pointerUpCount,             "Pointer up event is received by all-handlers component.");
+            Assert.Zero(handlerBasedListener1.pointerDraggedCount,        "Pointer dragged event is received by all-handlers component.");
+            Assert.Zero(handlerBasedListener1.speechCount,                "Speech event is received by all-handlers component.");
+
+            // No pointer clicked event:
+            Assert.Zero(handlerBasedListener2.pointerClickedCount,        "Pointer clicked event is received by speech-handler component.");
+            Assert.Zero(handlerBasedListener2.pointerDownCount,           "Pointer down event is received by speech-handler component.");
+            Assert.Zero(handlerBasedListener2.pointerUpCount,             "Pointer up event is received by speech-handler component.");
+            Assert.Zero(handlerBasedListener2.pointerDraggedCount,        "Pointer dragged event is received by speech-handler component.");
+            Assert.Zero(handlerBasedListener2.speechCount,                "Speech event is received by speech-handler component.");
+
+            /////
```

Comment:
```suggestion
```

Reply:
I think there are a few more of these (will probably just leave the rest as a find/replace exercise)